### PR TITLE
refactor(experience): support and apply modal loading state

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

This PR refactors the `ConfirmModal` component and its related hooks to support a loading state in modals and applies it to the experience flow.

Notably, it introduces a new callback-based usage pattern for confirm modals, providing greater flexibility in handling modal actions while offering improved user feedback during asynchronous operations.

Key changes:
1. Added support for modal loading state:
   - Introduced `isConfirmLoading` prop to `ModalProps` and `ConfirmModalState`
   - Updated `AcModal` and `MobileModal` components to handle the loading state
   - Confirm button automatically enters loading state during async confirm actions when using callback-style confirm modals

2. Implemented callback-based confirm modal usage:
   - Created `CallbackConfirmModalProps` type with `onConfirm` and `onCancel` callbacks
   - Added `showCallback` method to `ConfirmModalContextType`
   - Updated `ConfirmModalProvider` to handle both promise-based and callback-based modal actions
   - Automatically manages confirm button loading state when executing async `onConfirm` callbacks

3. Separated promise-based and callback-based hooks:
   - Introduced `usePromiseConfirmModal` for promise-based usage
   - Modified `useConfirmModal` for callback-based usage, supporting loading state during async operations

4. Removed unused modal render props:
   - Simplified `ModalContentRenderProps` by removing `confirm` and `cancel` props
   - Updated `ConfirmModalState` to use a simpler `ModalContent` type

5. Updated test cases to cover both promise-based and callback-based scenarios, including loading state tests

6. Refactored various components and hooks to use the new modal APIs:
   - Updated `use-identifier-error-alert`, `use-link-social-confirm-modal`, `use-register-flow-code-verification`, `use-sign-in-flow-code-verification`, `use-terms`, and several pages to use the appropriate modal hooks
   - Async operations in these scenarios now automatically trigger the confirm button's loading state

## Testing

https://github.com/user-attachments/assets/17a1ca40-9530-41ec-bde3-1d2aa819a4e1

https://github.com/user-attachments/assets/a6c1ecaf-16ef-4506-993c-196b4b528406

## Checklist

- [ ] `.changeset`
- [x] unit tests
- [ ] integration tests
- [x] necessary TSDoc comments

### CODE REVIEW

**Review by simeng-li — CHANGES_REQUESTED** [review]
need to discuss it offline

File: packages/experience/src/Providers/ConfirmModalProvider/index.tsx [hunk 1]
```diff
@@ -44,70 +42,68 @@ const defaultModalState: ConfirmModalState = {
 
 const ConfirmModalProvider = ({ children }: Props) => {
   const [modalState, setModalState] = useState<ConfirmModalState>(defaultModalState);
-
-  const resolver = useRef<(value: [result: boolean, data?: unknown]) => void>();
+  const [isConfirmLoading, setIsConfirmLoading] = useState(false);
+  const resolver = useRef<Resolver>();
```
Comment (simeng-li) [inline packages/experience/src/Providers/ConfirmModalProvider/index.tsx:46]: The point of using a resolver here is to make the confirmModal used a promise. If we go with the normal onConfirm/onCancell callback. This whole promisify logic is useless?

Comment (xiaoyijun) [reply packages/experience/src/Providers/ConfirmModalProvider/index.tsx]: updated.

**Review by simeng-li — APPROVED** [review]
overall LGTM

File: packages/experience/src/Providers/ConfirmModalProvider/index.tsx [hunk 2]
```diff
@@ -40,49 +47,86 @@ const defaultModalState: ConfirmModalState = {
...
+    },
+    []
+  );
 
-  const handleConfirm = useCallback((data?: unknown) => {
+  const handleConfirm = useCallback(async (data?: unknown) => {
+    if (callbackRef.current.onConfirm) {
+      setModalState((previous) => ({ ...previous, isConfirmLoading: true }));
+      await callbackRef.current.onConfirm();
+      setModalState((previous) => ({ ...previous, isConfirmLoading: false }));
```
[the remark this answers is not in the document]
Comment (xiaoyijun) [reply packages/experience/src/Providers/ConfirmModalProvider/index.tsx]: Updated.

File: packages/experience/src/Providers/ConfirmModalProvider/index.tsx [hunk 3]
```diff
@@ -40,49 +47,86 @@ const defaultModalState: ConfirmModalState = {
...
+    }
     resolver.current?.([true, data]);
     setModalState(defaultModalState);
   }, []);
 
   const handleCancel = useCallback((data?: unknown) => {
+    callbackRef.current.onCancel?.();
```
[the remark this answers is not in the document]
Comment (xiaoyijun) [reply packages/experience/src/Providers/ConfirmModalProvider/index.tsx]: Supported.

**Pull request merged**